src/kernel/arch/x86/irq.zig

```diff
@@ -120,4 +159,144 @@ pub fn init() void {
     openIrq(45, irq13);
     openIrq(46, irq14);
     openIrq(47, irq15);
+
+    log.logInfo("Done\n");
+
+    if (build_options.rt_test) runtimeTests();
+}
+
+extern fn testFunction0() void {}
+fn testFunction1(ctx: *arch.InterruptContext) void {}
+fn testFunction2(ctx: *arch.InterruptContext) void {}
+
+test "openIrq" {
+    idt.initTest();
+    defer idt.freeTest();
+
+    const index = u8(0);
+    const handler = testFunction0;
+    const ret: idt.IdtError!void = {};
+
+    idt.addTestParams("openInterruptGate", index, handler, ret);
+
+    openIrq(index, handler);
+}
+
+test "isValidIrq" {
+    comptime var i = 0;
+    inline while (i < NUMBER_OF_ENTRIES) : (i += 1) {
+        expectEqual(true, isValidIrq(i));
+    }
+
+    expectEqual(false, isValidIrq(200));
+}
+
+test "registerIrq re-register irq handler" {
+    // Set up
+    pic.initTest();
+    defer pic.freeTest();
+
+    pic.addTestParams("clearMask", u16(0));
+
+    // Pre testing
+    for (irq_handlers) |h1| {
+        if (h1) |_| {
+            std.debug.panic("Pre-testing failed");
+        }
+    }
+
+    // Call function
+    try registerIrq(0, testFunction1);
+    expectError(IrqError.IrqExists, registerIrq(0, testFunction2));
+
+    // Post testing
+    for (irq_handlers) |h, i| {
+        if (h) |temp| {
+            if (i != 0) {
+                std.debug.panic("Post-testing failed");
+            } else {
+                // Should still have the original handler
+                expectEqual(testFunction1, temp);
+            }
+        }
+    }
+
+    // Clean up
+    irq_handlers[0] = null;
+}
+
+test "registerIrq register irq handler" {
+    // Set up
+    pic.initTest();
+    defer pic.freeTest();
+
+    pic.addTestParams("clearMask", u16(0));
+
+    // Pre testing
+    for (irq_handlers) |h| {
+        if (h) |_| {
+            std.debug.panic("Pre-testing failed");
```

Comment:
This could be replaced with `expect(h == null)`

Reply:
Yes